components/ee/agent-smith/pkg/agent/agent.go

```diff
@@ -47,6 +51,7 @@ type Smith struct {
 
 	notifiedInfringements *lru.Cache
 	perfHandler           chan perfHandlerFunc
+	pidsMap               sync.Map
```

Comment:
We can probably expose a prometheus metrics to see how this grows, wdyt @csweichel ?